src/common/menu.rs

```diff
@@ -229,23 +235,25 @@ macro_rules! add_onoff_submenu {
     }
 }
 
-pub fn set_menu_from_url(s: &str) {
-    let base_url_len = "http://localhost/?".len();
-    let total_len = s.len();
+pub fn set_menu_from_url(s: &str) -> Result<(), Box<dyn std::error::Error>> {
+    let ss = s.split_once("?").unwrap().1;
+    let toggles_and_focus = match ss.split_once("#") {
+        Some(x) => x,
+        None => (ss, "")
+    };
```

Comment:
I think you can do 
```suggestion
    let (toggles, focus) = match ss.split_once("#") {
        Some(x) => x,
        None => (ss, "")
    };
```

https://doc.rust-lang.org/rust-by-example/primitives/tuples.html